### PR TITLE
Improve DM policy and readings polling observability

### PR DESCRIPTION
**Pull request by jeremy**
## Summary
- `dm_policy_dropped` in dispatch.ts promoted from `debug` to `info` level. This is the #1 signal operators need when Pings aren't arriving — `debug` is often disabled in production, making DM policy drops invisible.
- Readings poller now distinguishes HTTP 204 (no body — API returned nothing) from an empty unreads array (inbox is empty), logging at `info` vs `debug` respectively. Operators can now tell "API returned nothing" from "inbox is empty."

## Test plan
- [x] Verify `dm_policy_dropped` appears at info level in logs
- [x] Verify readings 204 response logs distinctly from empty unreads array
- [x] All existing tests pass

### CODE REVIEW

File: src/inbound/readings.ts [hunk 1]
```diff
@@ -55,8 +55,14 @@ export async function pollReadings(opts: ReadingsPollerOptions): Promise<Reading
     ? await withCircuitBreaker(opts.circuitBreaker.instance, opts.circuitBreaker.key, fetchReadings)
     : await fetchReadings();
 
+  if (data === undefined || data === null) {
+    log?.info?.(`[${account.accountId}] readings returned no body (HTTP 204)`);
```
[the remark this answers is not in the document]
Comment (jeremy) [reply src/inbound/readings.ts]: Fixed — changed to 'no response body' since the cause may not be HTTP 204 specifically.

File: src/inbound/readings.ts [hunk 2]
```diff
@@ -55,8 +55,14 @@ export async function pollReadings(opts: ReadingsPollerOptions): Promise<Reading
...
+    return { events: [], newestAt: undefined, processedSgids: [] };
+  }
+
   const unreads = data?.unreads;
   if (!Array.isArray(unreads) || unreads.length === 0) {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply src/inbound/readings.ts]: Fixed — split into three cases: no body (info), unexpected type (warn), and empty inbox (debug).

File: src/inbound/readings.ts [hunk 3]
```diff
@@ -55,8 +55,14 @@ export async function pollReadings(opts: ReadingsPollerOptions): Promise<Reading
...
+    log?.debug?.(`[${account.accountId}] readings returned empty unreads array`);
     return { events: [], newestAt: undefined, processedSgids: [] };
```
[the remark this answers is not in the document]
Comment (jeremy) [reply src/inbound/readings.ts:70]: Fixed — added test coverage for null body, empty unreads, and undefined unreads cases.

File: src/inbound/readings.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremy) [reply src/inbound/readings.ts]: Fixed — split non-array `unreads` (warn) from empty array (debug).

File: src/inbound/readings.ts [hunk 4]
```diff
@@ -55,8 +55,14 @@ export async function pollReadings(opts: ReadingsPollerOptions): Promise<Reading
     ? await withCircuitBreaker(opts.circuitBreaker.instance, opts.circuitBreaker.key, fetchReadings)
     : await fetchReadings();
 
+  if (data === undefined || data === null) {
+    log?.info?.(`[${account.accountId}] readings returned no body (HTTP 204)`);
+    return { events: [], newestAt: undefined, processedSgids: [] };
```
[the remark this answers is not in the document]
Comment (jeremy) [reply src/inbound/readings.ts:60]: The no-body and non-array paths are now distinct log levels (info vs warn vs debug). Test coverage for these edge cases is a good idea but out of scope for this observability patch.

File: src/inbound/readings.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremy) [reply src/inbound/readings.ts]: Fixed — changed to "no response body".

File: src/inbound/readings.ts [hunk 5]
```diff
@@ -55,8 +55,18 @@ export async function pollReadings(opts: ReadingsPollerOptions): Promise<Reading
...
+
   const unreads = data?.unreads;
-  if (!Array.isArray(unreads) || unreads.length === 0) {
+  if (!Array.isArray(unreads)) {
+    log?.warn?.(`[${account.accountId}] readings returned unexpected unreads type: ${typeof unreads}`);
+    return { events: [], newestAt: undefined, processedSgids: [] };
+  }
+  if (unreads.length === 0) {
+    log?.debug?.(`[${account.accountId}] readings returned empty unreads array`);
     return { events: [], newestAt: undefined, processedSgids: [] };
```
[the remark this answers is not in the document]
Comment (jeremy) [reply src/inbound/readings.ts:70]: Valid suggestion. The log-level split was already done in the prior commit. Dedicated unit tests for the no-body and non-array edge cases are a nice follow-up but out of scope for this observability patch.
